addons/mail/static/src/core/common/data_response_model.js

```diff
@@ -30,8 +36,13 @@ export class DataResponse extends Record {
     _autoResolve = false;
     /**
      * Promise that is resolved with the data when the data request is complete.
+     * @type {{
+     * promise: Promise<InstanceFields<DataResponse>>,
+     * resolve: (value: InstanceFields<DataResponse>) => void,
+     * reject: (reason?: any) => void
+     * }}
      */
-    _resultDef = new Deferred();
+    _resultPromiseWithResolvers = Promise.withResolvers();
```

Comment:
I would have named it `_resolver`. This name is too long and technical. In the context of `DataResponse`, I feel like `_resolver` is enough. Especially since the wording will spread now that deferred is deprecated in favor of withResolvers;